specs/0109-user-notifications-component.md

```diff
@@ -0,0 +1,55 @@
+* Feature Name: User Notifications Component
+* Created: 2021-02-25
+* Associated Bugs: [bug 1694911](https://bugzilla.mozilla.org/show_bug.cgi?id=1694911)
+
+# Summary
+
+Create a reusable component to display notifications to users as part of experiments and A/B testing.
+
+# Motivation
+
+We need a flexible way to notify users without writing ad-hoc code for every situation. Notifications should be manageable by admins without intervention from developers.
+
+Potential use cases:
+* We introduce a brand new feature, like string comments, and we want to make sure that users are aware of it. The infobar will include a short message and a link to documentation.
+* We want to promote the Pontoon add-on. The notification will include a button opening the AMO page to install the add-on.
+* We make changes to Pontoon terms and conditions, and want to require users to acknowledge them. The infobar will include a link to the updated terms, and a button to explicitly accept them.
+
+# Feature explanation
+
+When a logged in user opens Pontoon, we check against a list of notifications. If the user matches the necessary conditions, we display the notification.
+
+Information about each notification is stored in a table with the following fields:
+* Notification ID (automatically generated).
+* Identifier: short text identifier, e.g. `new_terms_2020`.
```

Comment:
OK, I see the value. Maybe we could pre-fill it from the title as a slug.

Reply:
Note: the title is optional, so generating from that wouldn't work. Since we're removing the modal option, should we remove the title too? Would the text be enough?